src/main.c

```diff
@@ -565,10 +565,31 @@ static void handle_sigusr2(int sock, short flags, void *arg)
 	}
 }
 
+/*
+ * Notify systemd that we are reloading, including a CLOCK_MONOTONIC timestamp in usec
+ * so that the program is compatible with a Type=notify-reload service.
+ *
+ * See https://www.freedesktop.org/software/systemd/man/latest/sd_notify.html
+ *
+ * Note that get_cached_time/get_time_usec from libusual uses gettimeofday
+ * instead of a monotonic clock, otherwise this function would be a single line:
+ *     sd_notifyf(0, "RELOADING=1\nMONOTONIC_USEC=%" PRIu64, get_cached_time());
+ */
+static void notify_reloading(void)
+{
+#ifdef USE_SYSTEMD
+	struct timespec ts;
+	usec_t usec;
+	clock_gettime(CLOCK_MONOTONIC, &ts);
+	usec = (usec_t)ts.tv_sec * USEC + (usec_t)ts.tv_nsec / (usec_t)1000;
+	sd_notifyf(0, "RELOADING=1\nMONOTONIC_USEC=%" PRIu64, usec);
```

Comment:
Does systemd ignore the MONOTONIC_USEC if the version is less than 253? AFAICS, yes. However, I didn't test.

Reply:
Yes, at least systemd 252 seems to be fine with this.